kedro/contrib/io/dask/dask_parquet.py

```diff
@@ -0,0 +1,131 @@
+# limitations under the License.
+
+"""``ParquetDaskDataSet`` is a data set used to load and save data to parquet files using Dask"""
+from copy import deepcopy
+from typing import Any, Dict
+
+import dask.dataframe as dd
+import fsspec
+from fsspec.utils import infer_storage_options
+
+from kedro.contrib.io import DefaultArgumentsMixIn
+from kedro.io.core import AbstractDataSet
+
+_PROTOCOL_DELIM = "://"
+
+
+class ParquetDaskDataSet(DefaultArgumentsMixIn, AbstractDataSet):
+    """``ParquetDaskDataSet`` loads and saves data to parquet file(s). It uses Dask
+    remote data services to handle the corresponding load and save operations:
+        https://docs.dask.org/en/latest/remote-data-services.html
+
+        Example (AWS S3):
+        ::
+
+            >>> from kedro.contrib.io.dask.dask_parquet import ParquetDaskDataSet
+            >>> import pandas as pd
+            >>> import dask.dataframe as dd
+            >>>
+            >>> data = pd.DataFrame({'col1': [1, 2], 'col2': [4, 5],
+            >>>                      'col3': [5, 6]})
+            >>> ddf = dd.from_pandas(data, npartitions=2)
+            >>>
+            >>> data_set = ParquetDaskDataSet(
+            >>>                         filepath="s3://bucket_name/path/to/folder",
+            >>>                         storage_options={
+            >>>                             'client_kwargs':
+            >>>                                 'aws_access_key_id': 'YOUR_KEY',
+            >>>                                 'aws_secret_access_key': 'YOUR SECRET'},
+            >>>                         save_args={"compression": "GZIP"})
+            >>> data_set.save(ddf)
+            >>> reloaded = data_set.load()
+            >>>
+            >>> assert ddf.compute().equals(reloaded.compute())
+    """
+
+    DEFAULT_SAVE_ARGS = {"write_index": False}
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        storage_options: Dict[str, Any] = None,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``ParquetDaskDataSet`` pointing to concrete
+        parquet files.
+
+        Args:
+            filepath: Path to a parquet file
+                parquet collection or the directory of a multipart parquet.
+            storage_options: Optional parameters to the backend file system driver,
+                such as credentials:
+                https://docs.dask.org/en/latest/remote-data-services.html#optional-parameters
+            load_args: Additional loading options `dask.dataframe.read_parquet`:
+                https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.read_parquet
+            save_args: Additional saving options for `dask.dataframe.to_parquet`:
+                https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.to_parquet
+        """
+        self._filepath = filepath
```

Comment:
Shouldn't it be an instance of `PurePosixPath`?

Reply:
Yes, I will change it. But unlike other datasets like `CSVDataSet`, `filepath` in this class should include protocol + delimiter + path, because `fsspec.filesystem` gets instantiated by dask API, not us. I was going to use `get_protocol_and_path()` and `get_filepath_str()`, but `get_filepath_str()` returns the full filepath with protocol for only HTTP, whereas I need to retrieve the full path with protocol for other protocol as well (S3, blob etc). So I made a similar func `get_filepath_str()` which return the full path with any protocol in this class.